### ISSUE TITLE
Handling PIDs/VIDs

### ISSUE THREAD

**Issue by Erovia** [qmk/qmk_firmware#8478]
We will need a solution to keep tab on the assigned unique PIDs/VIDs assigned to the keyboards.
Currently I know of 2 approaches:

1. A json file https://github.com/qmk/qmk_firmware/pull/8033
2. API https://yanfali.github.io/qmk_usb_usage/

**Comment by Duckle29** [comment]
To give a bit of info to help with this

The system proposed in #8033 is not really a storage of PID associations.

Instead it's a GH action to automatically assign a unique ID to new keyboards automatically once they're merged into master. This is done via a new QMK CLI command, and in order to ensure there aren't any collisions, this command will store the assigned PID *somewhere* for now in a json file.

This json file will only store IDs that have been assigned to the QMKVID via this script.

As I understand it, what 2 does is list *all* VID PID pairs used in QMK currently, and it does this by scraping the keyboard config files. 

In the future, we want to move forward to a system where we know all the VID:PID pairs used by QMK, and don't have to scrape for this. I think a major question is how to do this? A database, or specific json files have been proposed.

I should note that currently, the json file used by #8033 doesn't track anything but PIDs assigned by its own command, however because I've gotten some more info on the long-time goals, I'll tweak the storage format a bit so it'd be easier to track multiple VID:PIDs in that :)

I'll also have to look in to having the qmk cli command be the generic way to assign a VID:PID pair, even if you have your own, though I have to think about what that means a bit more
